app/controllers/ansible_roles_controller.rb

```diff
@@ -0,0 +1,44 @@
+class AnsibleRolesController < ::ApplicationController
+  include Foreman::Controller::AutoCompleteSearch
+
+  before_action :find_role, :only => [:destroy]
+  before_action :find_proxy, :only => [:import]
+  before_action :create_importer, :only => [:import, :confirm_import]
+
+  def index
+    @roles = resource_base.search_for(params[:search], :order => params[:order])
+      .paginate(:page => params[:page], :per_page => params[:per_page])
+  end
+
+  def destroy
+    if @role.destroy
+      process_success
+    else
+      process_error :object => @role
+    end
+  end
+
+  def import
+    @changed = @importer.import
```

Comment:
Please render local variables ( :locals => { :changed => @importer.import} ) in templates instead of relying in instance variables, it makes more difficult to change logic in templates